### PR TITLE
SE Messaging doc

### PR DESCRIPTION
**Pull request by danielkec**
Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

File: docs/mp/reactivemessaging/04_kafka.adoc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Kafka Connector
+:toc:
+:toc-placement: preamble
+:description: Reactive Messaging support for Kafka in Helidon MP
+:keywords: helidon, mp, messaging, kafka
+
```
Comment (ljamen) [inline docs/mp/reactivemessaging/04_kafka.adoc:24]: :h1Prefix: MP

File: docs/mp/reactivemessaging/04_kafka.adoc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+== Reactive Kafka Connector
+Connecting streams to Kafka via Reactive Messaging just can't be easier. 
+
```
Comment (ljamen) [inline docs/mp/reactivemessaging/04_kafka.adoc:27]: Connecting streams to Kafka with Reactive Messaging couldn't be easier.

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 3]
```diff
@@ -20,6 +20,96 @@
 :toc:
 :toc-placement: preamble
 :description: Reactive Messaging support in Helidon SE
-:keywords: helidon, mp, messaging
+:keywords: helidon, se, messaging
```
Comment (ljamen) [inline docs/se/reactivemessaging/01_introduction.adoc:23]: :h1Prefix: SE

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 4]
```diff
@@ -20,6 +20,96 @@
...
 
-== This page is Under Construction and will be available soon
+== Reactive Messaging
+
+Asynchronous messaging is a commonly used form of communication in the world of microservices.
+While it may seem natural in Helidon SE to start building your reactive streams directly
+by combining operators and connecting them to reactive APIs.
+There are repetitive use-cases you can use prepared tools for, and that is Helidon SE Reactive Messaging.
+
```
Comment (ljamen) [inline docs/se/reactivemessaging/01_introduction.adoc:31]: While its possible to start building your reactive streams directly by combining operators and connecting them to reactive APIs, with Helidon SE Reactive Messaging, you can now use prepared tools for repetitive use case scenarios .

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 5]
```diff
@@ -20,6 +20,96 @@
...
+>Consuming message TEST1 
+>Consuming message TEST2
+>Consuming message TEST3
+----
+
+=== Message
+Reactive Messaging in Helidon SE is using same concept of <<mp/reactivemessaging/02_message.adoc,Message wrapper>> as Microprofile messaging, only notable difference is that SE Messaging does almost no implicit or automatic acknowledgement due to _no magic_ phylosophy of Helidon SE. 
+
+Only exception to this are variants of methods `Messaging.Builder#listener` and `Messaging.Builder#processor` with consumer or function params, conviniently unwraping payload for you. After such implicit unwraping is not possible to do manual acknowledgement, therefore implicit ack before callback is executed is necessary.
```
Comment (ljamen) [inline docs/se/reactivemessaging/01_introduction.adoc:95]: Reactive Messaging in Helidon SE uses the same concept of messaging wrapping as Microprofile messaging. The only notable difference is that ....

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 6]
```diff
@@ -20,6 +20,96 @@
...
+>Consuming message TEST3
+----
+
+=== Message
+Reactive Messaging in Helidon SE is using same concept of <<mp/reactivemessaging/02_message.adoc,Message wrapper>> as Microprofile messaging, only notable difference is that SE Messaging does almost no implicit or automatic acknowledgement due to _no magic_ phylosophy of Helidon SE. 
+
+Only exception to this are variants of methods `Messaging.Builder#listener` and `Messaging.Builder#processor` with consumer or function params, conviniently unwraping payload for you. After such implicit unwraping is not possible to do manual acknowledgement, therefore implicit ack before callback is executed is necessary.   
+
+=== Connector
+Connector concept is a way for connecting <<Channel,channels>> to external sources. To make <<se/reactivemessaging/03_connector.adoc,creation and usage of connectors>> as easy and versatile as possible, Helidon SE Messaging uses same API for connectors like <<mp/reactivemessaging/01_introduction.adoc,MicroProfile Reactive Messaging>> does. This allows connectors to be usable in both flawors of Helidon with one limitation which is that connector has to be able to work without CDI.
```
Comment (ljamen) [inline docs/se/reactivemessaging/01_introduction.adoc:98]: typo: both flavors of Helidon...

File: docs/se/reactivemessaging/04_kafka.adoc [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Kafka Connector
+:toc:
+:toc-placement: preamble
+:description: Reactive Messaging support for Kafka in Helidon SE
+:keywords: helidon, se, messaging, kafka
```
Comment (ljamen) [inline docs/se/reactivemessaging/04_kafka.adoc:23]: :h1Prefix: SE

File: docs/se/reactivemessaging/04_kafka.adoc [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+== Reactive Kafka Connector
+Connecting streams to Kafka via Reactive Messaging just can't be easier. 
+
```
Comment (ljamen) [inline docs/se/reactivemessaging/04_kafka.adoc:27]: Connecting streams to Kafka with Reactive Messaging couldn't be easier.

File: docs/mp/reactivemessaging/04_kafka.adoc (hunk 1 above)
Comment (danielkec) [reply docs/mp/reactivemessaging/04_kafka.adoc:25]: thx fixed

File: docs/mp/reactivemessaging/04_kafka.adoc (hunk 2 above)
Comment (danielkec) [reply docs/mp/reactivemessaging/04_kafka.adoc:28]: thx fixed

File: docs/se/reactivemessaging/01_introduction.adoc (hunk 3 above)
Comment (danielkec) [reply docs/se/reactivemessaging/01_introduction.adoc:23]: thx fixed

File: docs/se/reactivemessaging/01_introduction.adoc (hunk 5 above)
Comment (danielkec) [reply docs/se/reactivemessaging/01_introduction.adoc]: thx fixed

File: docs/se/reactivemessaging/01_introduction.adoc (hunk 6 above)
Comment (danielkec) [reply docs/se/reactivemessaging/01_introduction.adoc]: thx fixed

File: docs/se/reactivemessaging/04_kafka.adoc (hunk 8 above)
Comment (danielkec) [reply docs/se/reactivemessaging/04_kafka.adoc:28]: thx fixed

File: docs/se/reactivemessaging/04_kafka.adoc (hunk 7 above)
Comment (danielkec) [reply docs/se/reactivemessaging/04_kafka.adoc:23]: thx fixed

File: docs/se/reactivemessaging/01_introduction.adoc (hunk 4 above)
Comment (danielkec) [reply docs/se/reactivemessaging/01_introduction.adoc:32]: thx fixed

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 9]
```diff
@@ -20,6 +20,113 @@
...
+Asynchronous messaging is a commonly used form of communication in the world of microservices.
+While its possible to start building your reactive streams directly by combining operators and
+connecting them to reactive APIs, with Helidon SE Reactive Messaging, you can now use prepared
+tools for repetitive use case scenarios .
+
+For example connecting your streams to external services usually requires a lot of boiler-plate
+code for configuration handling, backpressure propagation, acknowledgement and more.
+
+For such tasks there is a system of connectors, emitters and means to orchestrate them in Helidon,
+called *Reative Messaging*. It's basically an API for connecting and configuring
```
Comment (tjquinno) [inline docs/se/reactivemessaging/01_introduction.adoc:37]: Typo: "Reative" should be "Reactive"

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 10]
```diff
@@ -20,6 +20,113 @@
...
+You may wonder how *Reactive Messaging* relates to
+<<mp/reactivemessaging/01_introduction.adoc,MicroProfile Reactive Messaging>>.
+As the making of connectors or even configuring them can be repetitive task leading to
+the same results, Helidon SE Reactive Messaging supports very same configuration format
+for connectors as its MicroProfile counterpart does. Also, MP Connectors are reusable in
+Helidon SE Messaging with some limitation(there is no CDI in Helidon SE).
+All Messaging connectors in Helidon are made to be universally usable by Helidon MP and SE.
+
+=== Channel
+Channel is a named pair of `Publisher` and `Subscriber`, channels can be connected together by
```
Comment (tjquinno) [inline docs/se/reactivemessaging/01_introduction.adoc:49]: Maybe split into two sentences? 

From

and `Subscriber`, channels

to 

and `Subscriber`. Channels

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 11]
```diff
@@ -20,6 +20,113 @@
...
+        .start();
+
+>Consuming message TEST1 
+>Consuming message TEST2
+>Consuming message TEST3
+----
+
+=== Message
+Reactive Messaging in Helidon SE uses the same concept of
+<<mp/reactivemessaging/02_message.adoc,message wrapping>> as Microprofile messaging.
```
Comment (tjquinno) [inline docs/se/reactivemessaging/01_introduction.adoc:100]: MicroProfile should have an upper-case "P".

File: docs/se/reactivemessaging/01_introduction.adoc [hunk 12]
```diff
@@ -20,6 +20,113 @@
...
+The only notable difference is that SE Messaging does almost no implicit or automatic
+acknowledgement due to _no magic_ phylosophy of Helidon SE.
+
+Only exception to this are variants of methods `Messaging.Builder#listener` and
+`Messaging.Builder#processor` with consumer or function params, conviniently unwraping payload
```
Comment (tjquinno) [inline docs/se/reactivemessaging/01_introduction.adoc:105]: conviniently  should be conveniently

File: docs/se/reactivemessaging/01_introduction.adoc (hunk 9 above)
Comment (danielkec) [reply docs/se/reactivemessaging/01_introduction.adoc]: thx a lot , fixed

**Comment by danielkec** [comment]
Thx @tjquinno @tomas-langer all fixed

**Pull request merged**